### PR TITLE
Adding proxy settings to client setting

### PR DESCRIPTION
**Pull request by paigerube14**
Adding the same kubernetes config I added to cerberus in [PR](https://github.com/redhat-chaos/cerberus/pull/184) here, waiting for a private cluster to double check this. Will update if this doesn't work

### CODE REVIEW

File: kraken/kubernetes/client.py [hunk 1]
```diff
@@ -14,24 +16,43 @@
...
     global cli
     global batch_cli
+    global client_config
     global watch_resource
     global api_client
-    global dyn_client
     global custom_object_client
     try:
-        config.load_kube_config(kubeconfig_path)
+        """Initialize object and create clients from specified kubeconfig"""
```
Comment (sandrobonazzola) [inline kraken/kubernetes/client.py:30]: Sounds like this should be a comment rather than a string

File: kraken/kubernetes/client.py [hunk 2]
```diff
@@ -14,24 +16,43 @@
...
+        client_config = client.Configuration()
+        http_proxy = os.getenv("http_proxy", None)
+        """Proxy has auth header"""
```
Comment (sandrobonazzola) [inline kraken/kubernetes/client.py:33]: here as well

**Review by sandrobonazzola — CHANGES_REQUESTED** [review]
inline comments

**Review by chaitanyaenr — APPROVED** [review]
LGTM. Have not tested it though. Thanks.

**Review by ghost — CHANGES_REQUESTED** [review]

File: kraken/kubernetes/client.py [hunk 3]
```diff
@@ -14,24 +16,44 @@
...
-    global dyn_client
     global custom_object_client
     try:
-        config.load_kube_config(kubeconfig_path)
+        # Initialize object and create clients from specified kubeconfig
+        client_config = client.Configuration()
+        http_proxy = os.getenv("http_proxy", None)
+        # Proxy has auth header
+        if http_proxy and "@" in http_proxy:
+            proxy_auth = http_proxy.split("@")[0].split("//")[1]
```
[the remark this answers is not in the document]
Comment (paigerube14) [reply kraken/kubernetes/client.py]: Updated!

**Comment by chaitanyaenr** [comment]
@paigerube14 let's rebase the PR.

**Review by ghost — APPROVED** [review]

File: kraken/kubernetes/client.py [hunk 4]
```diff
@@ -15,23 +17,39 @@
...
     global custom_object_client
     try:
-        config.load_kube_config(kubeconfig_path)
+        # Initialize object and create clients from specified kubeconfig
+        client_config = client.Configuration()
+        http_proxy = os.getenv("http_proxy", None)
+        # Proxy has auth header
+        if http_proxy and "@" in http_proxy:
+            proxy_auth = urlparse(http_proxy)
+            auth_string = proxy_auth.username + ":" +  proxy_auth.password
```
[the remark this answers is not in the document]
Comment (paigerube14) [reply kraken/kubernetes/client.py:35]: Do you mean something like the below? I see a lot of references using that chardet python package but not sure if you have something else in mind

```
import chardet
the_encoding = chardet.detect('your string')['encoding']
string.decode(the_encoding)
```

**Comment by paigerube14** [comment]
closing as this functionality has moved to krkn-lib, will want to move the functionality over there

**Pull request closed without merging**